File: Vanilla/sources/zomboid/zombie/iso/SpriteDetails/IsoObjectType.java
package zombie.iso.SpriteDetails;

import java.util.HashMap;

public enum IsoObjectType {
   normal(0),
   jukebox(1),
   wall(2),
   stairsTW(3),
   stairsTN(4),
   stairsMW(5),
   stairsMN(6),
   stairsBW(7),
   stairsBN(8),
   UNUSED9(9),
   UNUSED10(10),
   doorW(11),
   doorN(12),
   lightswitch(13),
   radio(14),
   curtainN(15),
   curtainS(16),
   curtainW(17),
   curtainE(18),
   doorFrW(19),
   doorFrN(20),
   tree(21),
   windowFN(22),
   windowFW(23),
   UNUSED24(24),
   WestRoofB(25),
   WestRoofM(26),
   WestRoofT(27),
   isMoveAbleObject(28),
   MAX(29);

   private final int index;
   private static final HashMap fromStringMap = new HashMap();

   private IsoObjectType(int var3) {
      this.index = var3;
   }

   public int index() {
      return this.index;
   }

   public static IsoObjectType fromIndex(int var0) {
      return ((IsoObjectType[])IsoObjectType.class.getEnumConstants())[var0];
   }

   public static IsoObjectType FromString(String var0) {
      IsoObjectType var1 = (IsoObjectType)fromStringMap.get(var0);
      return var1 == null ? MAX : var1;
   }

   // $FF: synthetic method
   private static IsoObjectType[] $values() {
      return new IsoObjectType[]{normal, jukebox, wall, stairsTW, stairsTN, stairsMW, stairsMN, stairsBW, stairsBN, UNUSED9, UNUSED10, doorW, doorN, lightswitch, radio, curtainN, curtainS, curtainW, curtainE, doorFrW, doorFrN, tree, windowFN, windowFW, UNUSED24, WestRoofB, WestRoofM, WestRoofT, isMoveAbleObject, MAX};
   }

   static {
      IsoObjectType[] var0 = values();
      int var1 = var0.length;

      for(int var2 = 0; var2 < var1; ++var2) {
         IsoObjectType var3 = var0[var2];
         if (var3 == MAX) {
            break;
         }

         fromStringMap.put(var3.name(), var3);
      }

   }
}

File: Vanilla/sources/zdoc/media/lua/shared/Library/IsoObjectType.lua
---@class IsoObjectType : zombie.iso.SpriteDetails.IsoObjectType
---@field public normal IsoObjectType
---@field public jukebox IsoObjectType
---@field public wall IsoObjectType
---@field public stairsTW IsoObjectType
---@field public stairsTN IsoObjectType
---@field public stairsMW IsoObjectType
---@field public stairsMN IsoObjectType
---@field public stairsBW IsoObjectType
---@field public stairsBN IsoObjectType
---@field public UNUSED9 IsoObjectType
---@field public UNUSED10 IsoObjectType
---@field public doorW IsoObjectType
---@field public doorN IsoObjectType
---@field public lightswitch IsoObjectType
---@field public radio IsoObjectType
---@field public curtainN IsoObjectType
---@field public curtainS IsoObjectType
---@field public curtainW IsoObjectType
---@field public curtainE IsoObjectType
---@field public doorFrW IsoObjectType
---@field public doorFrN IsoObjectType
---@field public tree IsoObjectType
---@field public windowFN IsoObjectType
---@field public windowFW IsoObjectType
---@field public UNUSED24 IsoObjectType
---@field public WestRoofB IsoObjectType
---@field public WestRoofM IsoObjectType
---@field public WestRoofT IsoObjectType
---@field public isMoveAbleObject IsoObjectType
---@field public MAX IsoObjectType
---@field private index int
---@field private fromStringMap HashMap|Unknown|Unknown
IsoObjectType = {}

---@public
---@return int
function IsoObjectType:index() end

---Returns the enum constant of this type with the specified name.
---
---The string must match exactly an identifier used to declare an
---
---enum constant in this type.  (Extraneous whitespace characters are
---
---not permitted.)
---@public
---@param name String @the name of the enum constant to be returned.
---@return IsoObjectType @the enum constant with the specified name
function IsoObjectType:valueOf(name) end

---@public
---@param str String
---@return IsoObjectType
function IsoObjectType:FromString(str) end

---@public
---@param value int
---@return IsoObjectType
function IsoObjectType:fromIndex(value) end

---Returns an array containing the constants of this enum type, in
---
---the order they are declared.  This method may be used to iterate
---
---over the constants as follows:
---
---
---
---for (IsoObjectType c : IsoObjectType.values())
---
---    System.out.println(c);
---
---
---@public
---@return IsoObjectType[] @an array containing the constants of this enum type, in the order they are declared
function IsoObjectType:values() end
